---
layout: post
title:  'Varying non-linear dimensionality reduction on PCs'
author: Jamie Moore
jhed: jmoor185
categories: [ HW3 ]
image: homework/hw3/hw3_jmoor185.png
featured: false
---

I explored the question: If I perform non-linear dimensionality reduction on PCs, what happens when I vary how many PCs should I use?​

I am visualizing the non-linear dimensionality reduction on varying PCs of 15, and 30 of the variance between gene expression.
I am using the geometric primitive of points to represent each gene. I am using the geometric primitive of color saturation to encode
the variance in each gene expression per cell. I am also using the geometric primitive of positions to tell how close each cell type is
to each other. I am using the gesalt principle of proximity of putting the two different graphs next to each other. I am also using
the gesalt principle of similarity to show the differences of the two grpahs with varying number of PCs. My data visualization seeks 
to make more salient the relationship between more PCs in non-linear dimensionality reduction of gene expression.​

```{r}
data <- read.csv('~/Documents/genomicsDataVisualization/pikachu.csv.gz', row.names = 1)

library(ggplot2)
library(gridExtra)
library(Rtsne)
library(patchwork)

gexp <- data[, 6:ncol(data)]
topgene <- names(sort(apply(gexp, 2, var), decreasing=TRUE))
gexpfilter <- gexp[,topgene]
dim(gexpfilter)
pcs <- prcomp(gexpfilter)

emb1 <- Rtsne(pcs$x[,1:15], dims = 2, perplexity = 5)
emb2 <- Rtsne(pcs$x[,1:30], dims = 2, perplexity = 5)
df <- data.frame(pcs$x[,1:2], emb1=emb1$Y, emb2=emb2$Y, gene = rowSums(gexpfilter))
p1 <- ggplot(df) + geom_point(aes(x = emb1.1, y = emb1.2, col=gene)) + scale_colour_gradient2(low = 'blue', mid = 'lightgrey', high = 'cyan', midpoint = 600 ) + theme_classic()
p2 <- ggplot(df) + geom_point(aes(x = emb2.1, y = emb2.2, col=gene)) +scale_colour_gradient2(low = 'blue', mid = 'lightgrey', high = 'cyan', midpoint = 600) + theme_classic()
p1 + p2 + plot_annotation(tag_levels = 'a') + plot_layout(ncol = 1)

```
